addons/mail/models/mail_channel_rtc_session.py

```diff
@@ -0,0 +1,141 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from dateutil.relativedelta import relativedelta
+
+from odoo import api, fields, models
+
+
+class MailRtcSession(models.Model):
+    _name = 'mail.channel.rtc.session'
+    _description = 'Mail RTC session'
+
+    channel_id = fields.Many2one('mail.channel', index=True, required=True, ondelete='cascade')
+    channel_partner_id = fields.Many2one('mail.channel.partner', index=True, required=True, ondelete='cascade')
+    partner_id = fields.Many2one('res.partner', related='channel_partner_id.partner_id', string="Partner")
+    guest_id = fields.Many2one('mail.guest', related='channel_partner_id.guest_id')
+
+    write_date = fields.Datetime("Last Updated On", index=True)
+
+    is_screen_sharing_on = fields.Boolean(string="Is sharing the screen")
+    is_camera_on = fields.Boolean(string="Is sending user video")
+    is_muted = fields.Boolean(string="Is microphone muted")
+    is_deaf = fields.Boolean(string="Has disabled incoming sound")
+
+    def _update_and_broadcast(self, values):
+        """ Updates the session and notifies all members of the channel
+            of the change.
+        """
+        if not self.exists():
+            return
+        guest = self.env.context.get('guest')
+        if guest:
+            if guest != self.guest_id:
+                return
+        elif self.partner_id != self.env.user.partner_id:
+            return
+        self.write(values)
+        session_data = self._mail_rtc_session_format()
+        self.env['bus.bus'].sendone((self._cr.dbname, 'mail.channel', self.channel_id.id), {
+            'type': 'rtc_session_data_update',
+            'payload': {
+                'rtcSession': session_data,
+            },
+        })
+
+    @api.autovacuum
+    def _gc_inactive_sessions(self):
+        """ Garbage collect sessions that aren't active anymore,
+            this can happen when the server or the user's browser crash
+            or when the user's odoo session ends.
+        """
+        sessions = self.search([
+            ('write_date', '<', fields.Datetime.now() - relativedelta(days=1))
```

Comment:
Just a note but for me this is way too long functionally, but I know we had comments requesting otherwise for technical reasons so I'm ok to keep it like this for now and see how it goes.